src/main/java/util/TickingTimerManager.java

```diff
@@ -0,0 +1,69 @@
+package util;
+
+import ui.UI;
+
+/**
+ * This class is used to hold the TickingTimer instance and perform operations associated with it.
+ */
+public class TickingTimerManager {
+
+
+    public static final int BUFFER_TIME = 1; //To allow some additional time until next refresh.
+    private final TickingTimer timer;
+    public static final int REFRESH_PERIOD = 60;
+    public static final int APIQUOTA_BUFFER = 200;
+
+    public TickingTimerManager(TickingTimer timer) {
+        this.timer = timer;
+    }
+
+    public void startTimer() {
+        timer.start();
+    }
+
+    public void restartTimer() {
+        timer.restart();
+    }
+
+
+    /* This function updates the TickerTimer period, which is used to periodically refresh the issues.
+        * @param apiQuota : The remaining allowed api request until the next api request allowance renewal.
+        * @param remainingTime : The remaining time left until the next api request allowance renewal.
+        * @param lastApiCallsUsed : The amount of api used in the last api pull.
+    */
+    public int updateTickerTimer(int apiQuota, long remainingTime, int lastApiCallsUsed) {
```

Comment:
Try not to do multiple things in a method. The name says `update` but the method also calculates?